unsplahAPI/unsplahAPI/Authentification/Authentication/View/AuthenticationViewController.swift

```diff
@@ -9,42 +9,161 @@ import UIKit
+    @IBOutlet private weak var backgroundImageTopConstraint: NSLayoutConstraint!
+    @IBOutlet private weak var backgroundImageBottomConstraint: NSLayoutConstraint!
+    @IBOutlet private weak var underImageView: UIView!
+    @IBOutlet private weak var buttonsStackView: UIStackView! {
+        didSet {
+            for button in buttonsStackView.arrangedSubviews as? [UIButton] ?? [] {
+                button.layer.masksToBounds = true
+                button.layer.cornerRadius = button.bounds.height / 4
+            }
+        }
+    }
+    @IBOutlet private weak var createAccountButton: UIButton! {
+        didSet {
+            createAccountButton.setAttributedTitle(
+                NSAttributedString(
+                    string: "Create Account",
+                    attributes: [
+                        NSAttributedString.Key.font: DesignedSystemFonts.bodyBold,
+                        NSAttributedString.Key.foregroundColor: DesignedSystemColors.primaryText
+                    ]),
+                for: .normal
+            )
+        }
+    }
+    @IBOutlet private weak var logInButton: UIButton! {
+        didSet {
+            logInButton.setAttributedTitle(
+                NSAttributedString(
+                    string: "Log In",
+                    attributes: [
+                        NSAttributedString.Key.font: DesignedSystemFonts.bodyBold,
+                        NSAttributedString.Key.foregroundColor: DesignedSystemColors.primaryText
+                    ]),
+                for: .normal
+            )
+        }
+    }
+    @IBOutlet private weak var underButtonsLabel: UILabel! {
+        didSet {
+            underButtonsLabel.numberOfLines = 3
+            underButtonsLabel.attributedText = NSAttributedString(
+                string: "By proceeding you agree to some random terms of service and privacy policy",
+                attributes: [
+                    NSAttributedString.Key.font: DesignedSystemFonts.smallText,
+                    NSAttributedString.Key.foregroundColor: DesignedSystemColors.primaryInsignificantText
+            ]
+            )
+        }
+    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
-       userCancelable = viewModel?.user.$authenticationState.sink { [weak self] autheticationState in
+        userCancelable = viewModel?.user.$authenticationState.sink { [weak self] autheticationState in
             switch autheticationState {
             case .unauthenticated:
-                //TODO: show auth screen
+                // TODO: show auth screen
                 self?.authenticationProcess(result: .failure(AuthenticationErrors.unableToIdentifyUser))
             case .authenticated:
                 self?.authenticationProcess(result: .success(self?.viewModel?.user))
             case .authenticating:
-                //TODO: perform default process
+                // TODO: perform default process
                 print(autheticationState)
```

Comment:
I'm thinking this logic should be done in ViewModel as ViewController doesn't need to know about autenticationState or user at all, instead we can bind to success and perform navigation when we receive true.

```
class AuthenticationViewModel: AuthenticationViewModelProtocol {    
    
    var error: AnyPublisher<String, Never> {
        return errorSubject.eraseToAnyPublisher()
    }
    private let errorSubject = CurrentValueSubject<String, Never>("")
    
    var shouldNavigateToNextModule: AnyPublisher<Bool, Never> {
        return shouldNavigateToNextModuleSubject.eraseToAnyPublisher()
    }
    private let shouldNavigateToNextModuleSubject = CurrentValueSubject<Bool, Never>(false)
        
    func authenticate(username: String, password: String) {
        if error {
            errorSubject.value = "Failed to register the user"
        } else {
            shouldNavigateToNextModuleSubject.value = true
        }
    }
}
```

Reply:
I think, that the viewController should know, what he must present in case if viewModel receives user with the different authentification states